joblib/parallel.py

```diff
@@ -1281,6 +1284,14 @@ def _get_outputs(self, iterator, pre_dispatch):
             # the rest of the function does not call `_terminate_and_reset`
             # in finally.
             if dispatch_thread_id != threading.get_ident():
+                if not IS_PYPY:
+                    warnings.warn(
+                        "joblib.Parallel has detected an uncommon execution "
+                        "environment. joblib maintainers would like to know more "
+                        "about your usecase, please report this warning at "
+                        "https://github.com/joblib/joblib/issues ."
```

Comment:
```suggestion
                        "A generator produced by joblib.Parallel has been gc'ed in "
                        "an unexpected thread. This behavior should not cause major "
                        "issues but to make sure, please report this warning and "
                        "your use case at https://github.com/joblib/joblib/issues "
                        "so it can be investigated."
```